pip/commands/install.py

```diff
@@ -224,6 +236,14 @@ def run(self, options, args):
         if options.build_dir:
             options.build_dir = os.path.abspath(options.build_dir)
 
+        allowed_strategies = ["eager", "non-eager"]
```

Comment:
I don't think we'll ever add any new ones but this list should probably be kept close to the code actually performing the upgrade-or-not decision and be visible from here.

Please suggest what would be the correct/appropriate place to put this list of allowed upgrade strategies?